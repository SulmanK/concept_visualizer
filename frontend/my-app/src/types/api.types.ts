/**
 * API-related TypeScript interfaces for the Concept Visualizer application.
 */

/**
 * API Error response model
 */
export interface ApiError {
  status: number;
  message: string;
  details?: string;
}

/**
 * Common API response wrapper
 */
export interface ApiResponse<T> {
  data?: T;
  error?: ApiError;
  loading: boolean;
}

/**
 * Task status type
 */
export type TaskStatus = "pending" | "processing" | "completed" | "failed";

/**
 * Task response model
 */
export interface TaskResponse {
  id: string;
  task_id?: string;
  status: TaskStatus;
  type: "generate_concept" | "refine_concept";
  result_id?: string;
  error_message?: string;
  created_at: string;
  updated_at: string;
}

/**
 * Request model for concept generation
 */
export interface PromptRequest {
  logo_description: string;
  theme_description: string;
}

/**
 * Request model for concept refinement
 */
export interface RefinementRequest {
  original_image_url: string;
  logo_description?: string;
  theme_description?: string;
  refinement_prompt: string;
  preserve_aspects: string[];
}
