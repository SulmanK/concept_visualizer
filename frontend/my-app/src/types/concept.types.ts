/**
 * Concept-related TypeScript interfaces for the Concept Visualizer application.
 */

/**
 * Color palette model containing hex color codes
 */
export interface ColorPalette {
  primary: string;
  secondary: string;
  accent: string;
  background: string;
  text: string;
  additionalColors: string[];
}

/**
 * Color variation for a concept
 */
export interface ColorVariation {
  name: string;
  colors: string[];
  image_url: string;
  image_path?: string;
  description?: string;
}

/**
 * Response model for concept generation and refinement
 */
export interface GenerationResponse {
  // Backend API schema uses snake_case, while frontend uses camelCase
  // We need to support both naming conventions for compatibility
  
  // Primary fields that come from the backend
  prompt_id: string;
  logo_description: string;
  theme_description: string;
  created_at: string;
  image_url: string;
  color_palette: ColorPalette | null;
  generation_id?: string;
  variations?: ColorVariation[];
  original_image_url?: string | null;
  refinement_prompt?: string | null;
  id?: string; // ID of the saved concept in the database
  
  // These are aliases for frontend compatibility
  // They might not be in the API response
  imageUrl?: string;
  colorPalette?: ColorPalette | null;
  generationId?: string;
  createdAt?: string;
  originalImageUrl?: string | null;
}

/**
 * Stored concept data structure for recent concepts
 */
export interface StoredConcept {
  id: string;
  title?: string;
  logo_description: string;
  theme_description: string;
  image_url: string;
  image_path?: string;
  created_at: string;
  color_variations?: ColorVariation[];
} 