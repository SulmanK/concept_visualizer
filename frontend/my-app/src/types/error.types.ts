/**
 * Error-related TypeScript interfaces for the Concept Visualizer application.
 */

/**
 * Error categories enumeration
 */
export type ErrorCategory =
  | "network"
  | "auth"
  | "validation"
  | "rateLimit"
  | "api"
  | "server"
  | "unknown";

/**
 * Basic application debug info
 */
export interface DebugInfo {
  environment: string;
  apiBaseUrl: string;
  supabaseUrl: string;
  [key: string]: string | number | boolean;
}

/**
 * Validation error structure
 */
export interface ValidationError {
  field: string;
  message: string;
}

/**
 * Rate limit data structure
 */
export interface RateLimitData {
  limit: number;
  current: number;
  period: string;
  resetAfterSeconds: number;
}

/**
 * Common application error interface
 */
export interface ApplicationError {
  message: string;
  details?: string;
  category: ErrorCategory;
  status?: number;
  validationErrors?: ValidationError[];
  limit?: number;
  current?: number;
  period?: string;
  resetAfterSeconds?: number;
}
