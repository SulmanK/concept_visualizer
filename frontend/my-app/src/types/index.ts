/**
 * Re-export all types from domain-specific files
 */

// API Types
export * from "./api.types";

// Concept Types
export * from "./concept.types";

// UI Types
export * from "./ui.types";

// Form Types
export * from "./form.types";

/**
 * Form status type
 */
export type FormStatus =
  | "idle"
  | "submitting"
  | "initiating"
  | "pending"
  | "processing"
  | "success"
  | "error";
