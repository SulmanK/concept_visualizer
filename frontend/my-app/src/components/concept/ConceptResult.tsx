import React, { useState, useEffect } from "react";
import { GenerationResponse, ColorPalette } from "../../types";
import { Card } from "../ui/Card";
import { Button } from "../ui/Button";
import { ColorPalette as ColorPaletteComponent } from "../ui/ColorPalette";
import { logger } from "../../utils/logger";
import { devLog, devWarn, logError } from "../../utils/dev-logging";
import styles from "./ConceptResult.module.css";

export interface ConceptResultProps {
  /**
   * The generated concept data
   */
  concept: GenerationResponse;

  /**
   * Handler for requesting refinement
   */
  onRefineRequest?: () => void;

  /**
   * Handler for downloading the image
   */
  onDownload?: () => void;

  /**
   * Handler for selecting a color
   */
  onColorSelect?: (color: string) => void;

  /**
   * Optional color variations
   */
  variations?: Array<{
    name: string;
    colors: string[];
    image_url: string;
    description?: string;
  }>;

  /**
   * Handler for navigating to concept detail page
   */
  onExport?: (conceptId: string) => void;

  /**
   * The currently selected color
   */
  selectedColor?: string | null;

  /**
   * Handler for viewing concept details
   */
  onViewDetails?: () => void;
}

/**
 * Component for displaying generated concept results
 */
export const ConceptResult: React.FC<ConceptResultProps> = ({
  concept,
  onRefineRequest,
  onDownload,
  onColorSelect,
  variations = [],
  onExport,
  selectedColor = null,
  onViewDetails,
}) => {
  const [selectedVariation, setSelectedVariation] = useState<number | null>(
    null,
  );
  const [imageLoadErrors, setImageLoadErrors] = useState<
    Record<string, boolean>
  >({});

  useEffect(() => {
    // Reset selected variation when concept changes
    setSelectedVariation(null);
  }, [concept]);

  const handleImageError = (event: React.SyntheticEvent<HTMLImageElement>) => {
    const target = event.target as HTMLImageElement;
    devWarn(`Failed to load image: ${target.src}`);

    // Set a placeholder or fallback image
    target.src = "/images/placeholder-image.png";
    target.alt = "Image failed to load";
  };

  if (!concept || !concept.image_url) {
    return (
      <div className="p-6 text-center">
        <p className="text-gray-500">No concept data available</p>
      </div>
    );
  }

  // Format the date - just use created_at
  const createdDate = concept.created_at
    ? new Date(concept.created_at).toLocaleString()
    : "";

  const getFileName = (url: string): string => {
    try {
      // Extract filename from URL
      const parts = url.split("/");
      const filename = parts[parts.length - 1].split("?")[0]; // Remove query params if present

      // If the filename looks valid, use it
      if (filename && filename.includes(".")) {
        return filename;
      }
    } catch (error) {
      logError(error, "Error extracting filename from URL");
    }

    // Fallback filename
    return `concept-${concept.id || "image"}.png`;
  };

  const handleDownload = () => {
    if (onDownload) {
      onDownload();
      return;
    }

    // Use the formatted URL of the current image (variation or original)
    const imageUrl = getCurrentImageUrl();
    const filename = getFileName(imageUrl);

    // Create a link element and trigger download
    const link = document.createElement("a");
    link.href = imageUrl;
    link.download = filename;
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
  };

  const getCurrentPalette = (): ColorPalette | string[] => {
    if (
      selectedVariation !== null &&
      variations &&
      variations[selectedVariation]
    ) {
      return variations[selectedVariation].colors;
    }

    return concept.color_palette || [];
  };

  const getOriginalImageUrl = (): string => {
    return concept.image_url || "/images/placeholder-image.png";
  };

  const getCurrentImageUrl = () => {
    // If a variation is selected, use its image_url
    if (
      selectedVariation !== null &&
      variations &&
      variations[selectedVariation]
    ) {
      return (
        variations[selectedVariation].image_url ||
        "/images/placeholder-image.png"
      );
    }

    // Otherwise use the original concept image
    return concept.image_url || "/images/placeholder-image.png";
  };

  const getCurrentVariationName = () => {
    if (
      selectedVariation !== null &&
      variations &&
      variations[selectedVariation]
    ) {
      return variations[selectedVariation].name;
    }

    return "Original";
  };

  const handleColorSelect = (color: string) => {
    if (onColorSelect) {
      onColorSelect(color);
    }
  };

  const getCurrentPaletteColors = () => {
    const palette = getCurrentPalette();
    return Array.isArray(palette)
      ? palette
      : (Object.values(palette as ColorPalette).filter(
          (val) => typeof val === "string",
        ) as string[]);
  };

  const getImageElements = () => {
    return variations.map((variation) => variation.image_url);
  };

  // Function to create a proper ColorPalette object from an array or existing object
  const createColorPaletteObject = (
    colors: string[] | ColorPalette,
  ): ColorPalette => {
    // If it's already a ColorPalette object with the expected properties
    if (
      typeof colors === "object" &&
      !Array.isArray(colors) &&
      "primary" in colors &&
      "secondary" in colors
    ) {
      return colors as ColorPalette;
    }

    // Convert array to ColorPalette object
    const colorArray = Array.isArray(colors)
      ? colors
      : ((typeof colors === "object"
          ? Object.values(colors).filter((val) => typeof val === "string")
          : []) as string[]);

    const palette: ColorPalette = {
      primary: colorArray[0] || "#4F46E5",
      secondary: colorArray[1] || "#818CF8",
      accent: colorArray[2] || "#C7D2FE",
      background: colorArray[3] || "#EEF2FF",
      text: colorArray[4] || "#312E81",
      additionalColors: [],
    };

    // Add any additional colors
    if (colorArray.length > 5) {
      palette.additionalColors = colorArray.slice(5);
    }

    return palette;
  };

  // Get the current palette as a proper ColorPalette object
  const currentPalette = createColorPaletteObject(getCurrentPalette());

  return (
    <div className={styles.container}>
      {/* Header with title and date */}
      <div className={styles.pageHeader}>
        <h2 className={styles.title}>Generated Concept</h2>
        {createdDate && <span className={styles.date}>{createdDate}</span>}
      </div>

      {/* Main concept image */}
      <div className={styles.section}>
        <div className={styles.sectionTitle}>{getCurrentVariationName()}</div>
        <div className={styles.mainImageContainer}>
          <img
            src={getCurrentImageUrl()}
            alt={`Generated concept - ${getCurrentVariationName()}`}
            className={styles.image}
            onError={handleImageError}
          />
        </div>
      </div>

      {/* Color palette section */}
      <div className={styles.section}>
        <h3 className={styles.sectionTitle}>Color Palette</h3>

        <div className={styles.card}>
          <ColorPaletteComponent
            palette={currentPalette}
            showLabels={true}
            selectable={!!onColorSelect}
            selectedColor={selectedColor || undefined}
            onColorSelect={(color) => handleColorSelect(color)}
            size="md"
            className="flex justify-center"
          />
        </div>
      </div>

      {/* Variations */}
      {variations && variations.length > 0 && (
        <div className={styles.section}>
          <h3 className={styles.sectionTitle}>Variations</h3>
          <div className={styles.variationsGrid}>
            {/* Original version */}
            <div
              className={`${styles.variationCard} ${
                selectedVariation === null ? styles.variationCardSelected : ""
              } hover:shadow-md`}
              onClick={() => setSelectedVariation(null)}
            >
              <div className={styles.variationHeader}>
                <span className={styles.variationTitle}>Original</span>
              </div>
              <div className={styles.variationContent}>
                <img
                  src={getOriginalImageUrl()}
                  alt="Original concept"
                  className={styles.variationImage}
                  onError={handleImageError}
                />
                <div className={styles.colorDotsContainer}>
                  {concept.color_palette && Array.isArray(concept.color_palette)
                    ? concept.color_palette
                        .slice(0, 5)
                        .map((color, idx) => (
                          <div
                            key={`original-color-${idx}`}
                            className={styles.colorDot}
                            style={{ backgroundColor: color }}
                          />
                        ))
                    : concept.color_palette &&
                      typeof concept.color_palette === "object"
                    ? Object.values(concept.color_palette)
                        .filter((val) => typeof val === "string")
                        .slice(0, 5)
                        .map((color, idx) => (
                          <div
                            key={`original-color-${idx}`}
                            className={styles.colorDot}
                            style={{ backgroundColor: color as string }}
                          />
                        ))
                    : null}
                </div>
              </div>
            </div>

            {/* Other variations */}
            {variations.map((variation, index) => (
              <div
                key={`variation-${index}`}
                className={`${styles.variationCard} ${
                  selectedVariation === index
                    ? styles.variationCardSelected
                    : ""
                } hover:shadow-md`}
                onClick={() => setSelectedVariation(index)}
              >
                <div className={styles.variationHeader}>
                  <span className={styles.variationTitle}>
                    {variation.name}
                  </span>
                </div>
                <div className={styles.variationContent}>
                  <img
                    src={variation.image_url}
                    alt={`Variation: ${variation.name}`}
                    className={styles.variationImage}
                    onError={handleImageError}
                  />
                  <div className={styles.colorDotsContainer}>
                    {variation.colors.slice(0, 5).map((color, idx) => (
                      <div
                        key={`var-${index}-color-${idx}`}
                        className={styles.colorDot}
                        style={{ backgroundColor: color }}
                      />
                    ))}
                  </div>
                </div>
              </div>
            ))}
          </div>
        </div>
      )}

      {/* Action buttons */}
      <div className={styles.actionsContainer}>
        {onViewDetails && (
          <Button variant="primary" onClick={onViewDetails}>
            View Details
          </Button>
        )}

        {onRefineRequest && (
          <Button variant="outline" onClick={onRefineRequest}>
            Refine Concept
          </Button>
        )}

        <Button variant="outline" onClick={handleDownload}>
          Download
        </Button>

        {onExport && concept.id && (
          <Button
            variant="outline"
            onClick={() => onExport(concept.id as string)}
          >
            Export
          </Button>
        )}
      </div>
    </div>
  );
};
