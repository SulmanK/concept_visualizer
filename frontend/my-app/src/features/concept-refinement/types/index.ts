/**
 * Types specific to the Concept Refinement feature
 */
import { GenerationResponse } from '../../../types/concept.types';
import { FormStatus } from '../../../types/ui.types';

/**
 * Original concept model
 */
export interface OriginalConcept {
  imageUrl: string;
  logoDescription?: string;
  themeDescription?: string;
}

/**
 * Props for the RefinementForm component
 */
export interface RefinementFormProps {
  originalImageUrl: string;
  onSubmit: (
    refinementPrompt: string,
    logoDescription: string,
    themeDescription: string,
    preserveAspects: string[]
  ) => void;
  status: FormStatus;
  error?: string;
  onCancel: () => void;
  initialLogoDescription?: string;
  initialThemeDescription?: string;
}

/**
 * Props for the RefinementActions component
 */
export interface RefinementActionsProps {
  onReset: () => void;
  onCreateNew: () => void;
}

/**
 * Props for the ComparisonView component
 */
export interface ComparisonViewProps {
  originalConcept: OriginalConcept;
  refinedConcept: GenerationResponse;
  onColorSelect?: (color: string) => void;
  onRefineRequest?: () => void;
} 