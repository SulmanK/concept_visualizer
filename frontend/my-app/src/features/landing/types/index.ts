/**
 * Types specific to the Landing page feature
 */
import { GenerationResponse } from '../../../types/concept.types';
import { FormStatus } from '../../../types/ui.types';

/**
 * Step model for the How It Works section
 */
export interface HowItWorksStep {
  number: number;
  title: string;
  description: string;
}

/**
 * Props for the HowItWorks component
 */
export interface HowItWorksProps {
  steps: HowItWorksStep[];
  onGetStarted: () => void;
}

/**
 * Recent concept model for display in RecentConcepts section
 */
export interface RecentConcept {
  id: string;
  title: string;
  description: string;
  colors: string[];
  gradient: {
    from: string;
    to: string;
  };
  initials: string;
}

/**
 * Props for the ResultsSection component
 */
export interface ResultsSectionProps {
  result: GenerationResponse;
  onReset: () => void;
  selectedColor: string | null;
  onColorSelect: (color: string) => void;
}

/**
 * Props for the ConceptFormSection component
 */
export interface ConceptFormSectionProps {
  onSubmit: (logoDescription: string, themeDescription: string) => void;
  status: FormStatus;
  error?: string;
  onReset: () => void;
}

/**
 * Props for the RecentConceptsSection component
 */
export interface RecentConceptsSectionProps {
  concepts: RecentConcept[];
  onEdit: (id: string) => void;
  onViewDetails: (id: string) => void;
} 