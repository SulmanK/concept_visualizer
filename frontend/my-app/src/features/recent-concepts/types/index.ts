/**
 * Types specific to the Recent Concepts feature
 */
import { StoredConcept, ColorVariation } from '../../../types/concept.types';

/**
 * Props for ConceptCard component
 */
export interface ConceptCardProps {
  concept: StoredConcept;
}

/**
 * Context state for recent concepts
 */
export interface RecentConceptsState {
  recentConcepts: StoredConcept[];
  loadingConcepts: boolean;
  errorLoadingConcepts: string | null;
  refreshConcepts: () => void;
} 