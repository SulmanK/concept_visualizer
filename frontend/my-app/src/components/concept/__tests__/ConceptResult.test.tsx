import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { ConceptResult } from '../ConceptResult';
import { GenerationResponse } from '../../../types';

// Simple mock for window.open to test download functionality
window.open = jest.fn();

describe('ConceptResult Component', () => {
  // Mock handlers
  const mockRefineRequest = jest.fn();
  const mockDownload = jest.fn();
  const mockColorSelect = jest.fn();
  
  // Sample concept data for testing
  const sampleConcept: GenerationResponse = {
    image_url: 'https://example.com/generated-concept.png',
    id: 'concept-123',
    created_at: '2023-04-15T12:30:45Z',
    prompt_id: 'prompt-123',
    logo_description: 'A modern tech logo with blue gradient',
    theme_description: 'Modern, professional, tech-focused design',
    color_palette: {
      primary: '#4F46E5',
      secondary: '#60A5FA',
      accent: '#EEF2FF',
      background: '#FFFFFF',
      text: '#1E293B',
      additionalColors: ['#818CF8', '#6366F1'],
    },
  };
  
  // Reset mocks before each test
  beforeEach(() => {
    jest.clearAllMocks();
  });
  
  // Test for no concept data
  test('renders placeholder when no concept data is available', () => {
    render(
      <ConceptResult 
        concept={null as any}
        onRefineRequest={mockRefineRequest}
        onColorSelect={mockColorSelect}
      />
    );
    
    const placeholder = screen.getByText('No concept data available');
    expect(placeholder).toBeInTheDocument();
  });
  
  // Basic rendering test
  test('renders the concept image and color palette', () => {
    render(
      <ConceptResult 
        concept={sampleConcept}
        onRefineRequest={mockRefineRequest}
        onColorSelect={mockColorSelect}
      />
    );
    
    // Check for the generated image
    const conceptImage = screen.getByAltText(/Generated concept/i);
    expect(conceptImage).toBeInTheDocument();
    expect(conceptImage).toHaveAttribute('src', sampleConcept.image_url);
    
    // Check for the color palette heading
    const paletteHeading = screen.getByText('Color Palette');
    expect(paletteHeading).toBeInTheDocument();
    
    // Check for buttons
    const refineButton = screen.getByRole('button', { name: /Refine Concept/i });
    expect(refineButton).toBeInTheDocument();
  });
  
  // Refine button test
  test('calls onRefineRequest when Refine Concept button is clicked', () => {
    render(
      <ConceptResult 
        concept={sampleConcept}
        onRefineRequest={mockRefineRequest}
        onColorSelect={mockColorSelect}
      />
    );
    
    // Find and click the refine button
    const refineButton = screen.getByRole('button', { name: /Refine Concept/i });
    fireEvent.click(refineButton);
    
    // Verify that the refine handler was called
    expect(mockRefineRequest).toHaveBeenCalled();
  });
  
  // Download button with custom handler test
  test('calls onDownload when Download button is clicked and onDownload is provided', () => {
    render(
      <ConceptResult 
        concept={sampleConcept}
        onRefineRequest={mockRefineRequest}
        onDownload={mockDownload}
        onColorSelect={mockColorSelect}
      />
    );
    
    // Find and click the download button
    const downloadButton = screen.getByRole('button', { name: /Download/i });
    fireEvent.click(downloadButton);
    
    // Verify that the download handler was called
    expect(mockDownload).toHaveBeenCalled();
  });
  
  // Test without refine button
  test('does not render Refine Concept button when onRefineRequest is not provided', () => {
    render(
      <ConceptResult 
        concept={sampleConcept}
        onColorSelect={mockColorSelect}
      />
    );
    
    // Check that the refine button is not rendered
    const refineButton = screen.queryByRole('button', { name: /Refine Concept/i });
    expect(refineButton).not.toBeInTheDocument();
    
    // But the download button should still be there
    const downloadButton = screen.getByRole('button', { name: /Download/i });
    expect(downloadButton).toBeInTheDocument();
  });
  
  // Snapshot test
  test('matches snapshot', () => {
    const { container } = render(
      <ConceptResult 
        concept={sampleConcept}
        onRefineRequest={mockRefineRequest}
        onColorSelect={mockColorSelect}
      />
    );
    
    expect(container.firstChild).toMatchSnapshot();
  });
}); 