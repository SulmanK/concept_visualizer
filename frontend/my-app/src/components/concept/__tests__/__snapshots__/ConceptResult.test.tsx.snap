// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConceptResult Component > matches snapshot 1`] = `
<div
  class="_container_24d64d"
>
  <div
    class="_pageHeader_24d64d"
  >
    <h2
      class="_title_24d64d"
    >
      Generated Concept
    </h2>
    <span
      class="_date_24d64d"
    >
      4/15/2023, 8:30:45 AM
    </span>
  </div>
  <div
    class="_section_24d64d"
  >
    <div
      class="_sectionTitle_24d64d"
    >
      Original
    </div>
    <div
      class="_mainImageContainer_24d64d"
    >
      <img
        alt="Generated concept - Original"
        class="_image_24d64d"
        src="https://example.com/generated-concept.png"
      />
    </div>
  </div>
  <div
    class="_section_24d64d"
  >
    <h3
      class="_sectionTitle_24d64d"
    >
      Color Palette
    </h3>
    <div
      class="_card_24d64d"
    >
      <div
        class="flex flex-col space-y-4 flex justify-center"
      >
        <div
          class="flex flex-wrap gap-3"
        >
          <div
            class="flex flex-col items-center"
          >
            <div
              class="
                h-8 w-8 rounded-full shadow-sm border border-dark-200
                cursor-pointer hover:scale-110 transition-transform

              "
              style="background-color: rgb(79, 70, 229);"
              title="Primary: #4F46E5"
            />
            <span
              class="mt-1 text-xs text-dark-600"
            >
              Primary
            </span>
          </div>
          <div
            class="flex flex-col items-center"
          >
            <div
              class="
                h-8 w-8 rounded-full shadow-sm border border-dark-200
                cursor-pointer hover:scale-110 transition-transform

              "
              style="background-color: rgb(96, 165, 250);"
              title="Secondary: #60A5FA"
            />
            <span
              class="mt-1 text-xs text-dark-600"
            >
              Secondary
            </span>
          </div>
          <div
            class="flex flex-col items-center"
          >
            <div
              class="
                h-8 w-8 rounded-full shadow-sm border border-dark-200
                cursor-pointer hover:scale-110 transition-transform

              "
              style="background-color: rgb(238, 242, 255);"
              title="Accent: #EEF2FF"
            />
            <span
              class="mt-1 text-xs text-dark-600"
            >
              Accent
            </span>
          </div>
          <div
            class="flex flex-col items-center"
          >
            <div
              class="
                h-8 w-8 rounded-full shadow-sm border border-dark-200
                cursor-pointer hover:scale-110 transition-transform

              "
              style="background-color: rgb(255, 255, 255);"
              title="Background: #FFFFFF"
            />
            <span
              class="mt-1 text-xs text-dark-600"
            >
              Background
            </span>
          </div>
          <div
            class="flex flex-col items-center"
          >
            <div
              class="
                h-8 w-8 rounded-full shadow-sm border border-dark-200
                cursor-pointer hover:scale-110 transition-transform

              "
              style="background-color: rgb(30, 41, 59);"
              title="Text: #1E293B"
            />
            <span
              class="mt-1 text-xs text-dark-600"
            >
              Text
            </span>
          </div>
        </div>
        <div
          class="mt-2"
        >
          <p
            class="text-sm text-dark-600 mb-1"
          >
            Additional Colors
          </p>
          <div
            class="flex flex-wrap gap-2"
          >
            <div
              class="flex flex-col items-center"
            >
              <div
                class="
                    h-8 w-8 rounded-full shadow-sm border border-dark-200
                    cursor-pointer hover:scale-110 transition-transform

                  "
                style="background-color: rgb(129, 140, 248);"
                title="Additional color: #818CF8"
              />
              <span
                class="mt-1 text-xs text-dark-600"
              >
                A
                1
              </span>
            </div>
            <div
              class="flex flex-col items-center"
            >
              <div
                class="
                    h-8 w-8 rounded-full shadow-sm border border-dark-200
                    cursor-pointer hover:scale-110 transition-transform

                  "
                style="background-color: rgb(99, 102, 241);"
                title="Additional color: #6366F1"
              />
              <span
                class="mt-1 text-xs text-dark-600"
              >
                A
                2
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="_actionsContainer_24d64d"
  >
    <button
      class="inline-flex items-center justify-center font-medium focus:outline-none focus:ring-2 focus:ring-primary/30 focus:ring-offset-2 disabled:opacity-60 disabled:cursor-not-allowed transition-all duration-200 border border-indigo-300 text-indigo-700 bg-white hover:bg-indigo-50 hover:text-primary-dark text-sm px-4 py-2 rounded-lg transform scale-100 hover:scale-[1.02] "
      type="button"
    >
      Refine Concept
    </button>
    <button
      class="inline-flex items-center justify-center font-medium focus:outline-none focus:ring-2 focus:ring-primary/30 focus:ring-offset-2 disabled:opacity-60 disabled:cursor-not-allowed transition-all duration-200 border border-indigo-300 text-indigo-700 bg-white hover:bg-indigo-50 hover:text-primary-dark text-sm px-4 py-2 rounded-lg transform scale-100 hover:scale-[1.02] "
      type="button"
    >
      Download
    </button>
  </div>
</div>
`;
